Update ReactiveUI/Splat dependencies and re-enable NuGet signature verification (NU3012 revoked cert)

## Background

CI restore currently fails with **NU3012** because the code-signing certificate used for the **ReactiveUI / Splat** NuGet packages has been **revoked**:

```
error NU3012: Package ''ReactiveUI 23.1.8'' from source ''https://api.nuget.org/v3/index.json'':
The author primary signature found a chain building issue: Revoked: certificate revoked
```

Affected packages: `ReactiveUI` / `ReactiveUI.Blazor` (23.1.8), `Splat`, `Splat.Builder`, `Splat.Core`, `Splat.Logging` (19.3.1).

The GitHub-hosted Ubuntu runners enforce NuGet package signature verification (`DOTNET_NUGET_SIGNATURE_VERIFICATION=true`), so a revoked signing cert becomes a hard restore failure and blocks all builds.

## Immediate mitigation (done)

PR #<fill-in> disables enforced signature verification (`DOTNET_NUGET_SIGNATURE_VERIFICATION=false`) on the restore steps in the `CodeQuality`, `codeql` and `nuget-reference-check` workflows and in the Dockerfile build stage. This unblocks CI but turns off signature verification for restore.

## Longer-term fix (this issue)

- [ ] Update `ReactiveUI` / `ReactiveUI.Blazor` / `Splat*` to a version re-signed with a valid certificate once the maintainers republish (track upstream: reactiveui/ReactiveUI and reactivemarbles/Splat). - version 23.2.28 
- [ ] Once dependencies are on validly-signed versions, **revert** the `DOTNET_NUGET_SIGNATURE_VERIFICATION=false` workaround so package signature verification is re-enabled in CI and the Docker build.

## Acceptance criteria

- CI restore succeeds with signature verification re-enabled (workaround removed).
- ReactiveUI/Splat references point to non-revoked, validly-signed package versions.

--- a/.github/workflows/CodeQuality.yml
+++ b/.github/workflows/CodeQuality.yml
@@ -41,10 +41,6 @@ jobs:
         run: dotnet nuget add source https://nuget.devexpress.com/api -n DXFeed -u DevExpress -p $DEVEXPRESS_NUGET_KEY --store-password-in-clear-text
 
       - name: Restore dependencies
-        env:
-          # Workaround for NU3012: the ReactiveUI/Splat NuGet signing certificate has been revoked.
-          # Disable enforced package signature verification until the packages are updated. See issue for the long-term fix.
-          DOTNET_NUGET_SIGNATURE_VERIFICATION: false
         run: dotnet restore COMETwebapp.sln
 
       - name: Sonarqube Begin

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -50,10 +50,6 @@ jobs:
       run: dotnet nuget add source https://nuget.devexpress.com/api -n DXFeed -u DevExpress -p $DEVEXPRESS_NUGET_KEY --store-password-in-clear-text
 
     - name: Install dependencies
-      env:
-        # Workaround for NU3012: the ReactiveUI/Splat NuGet signing certificate has been revoked.
-        # Disable enforced package signature verification until the packages are updated. See issue for the long-term fix.
-        DOTNET_NUGET_SIGNATURE_VERIFICATION: false
       run: dotnet restore COMETwebapp.sln
 
     - name: Build

--- a/.github/workflows/nuget-reference-check.yml
+++ b/.github/workflows/nuget-reference-check.yml
@@ -33,10 +33,6 @@ jobs:
       run: dotnet nuget add source https://nuget.devexpress.com/api -n DXFeed -u DevExpress -p $DEVEXPRESS_NUGET_KEY --store-password-in-clear-text
 
     - name: Restore dependencies
-      env:
-        # Workaround for NU3012: the ReactiveUI/Splat NuGet signing certificate has been revoked.
-        # Disable enforced package signature verification until the packages are updated. See issue for the long-term fix.
-        DOTNET_NUGET_SIGNATURE_VERIFICATION: false
       run: dotnet restore COMETwebapp.sln
 
     - name: Build

--- a/COMET.Web.Common.Tests/Components/Applications/SingleIterationApplicationTemplateTestFixture.cs
+++ b/COMET.Web.Common.Tests/Components/Applications/SingleIterationApplicationTemplateTestFixture.cs
@@ -175,11 +175,8 @@ namespace COMET.Web.Common.Tests.Components.Applications
 
             var navigationManager = this.context.Services.GetService<NavigationManager>();
 
-            Assert.Multiple(() =>
-            {
-                Assert.That(navigationManager.Uri, Is.EqualTo("http://localhost/"));
-                this.viewModel.Verify(x => x.OnThingSelect(this.openIterations.Items[0]), Times.Exactly(2));
-            });
+            renderer.WaitForAssertion(() => Assert.That(navigationManager.Uri, Is.EqualTo("http://localhost/")));
+            this.viewModel.Verify(x => x.OnThingSelect(this.openIterations.Items[0]), Times.Exactly(2));
 
             this.viewModel.Setup(x => x.SelectedThing).Returns(this.openIterations.Items[0]);
             renderer.Instance.SetCorrectUrl();
@@ -220,9 +217,10 @@ namespace COMET.Web.Common.Tests.Components.Applications
             this.viewModel.Setup(x => x.SelectedThing).Returns((Iteration)null);
             renderer.Instance.SetCorrectUrl();
 
+            renderer.WaitForAssertion(() => Assert.That(navigationManager.Uri, Is.EqualTo("http://localhost/")));
+
             Assert.Multiple(() =>
             {
-                Assert.That(navigationManager.Uri, Is.EqualTo("http://localhost/"));
                 Assert.That(() => renderer.FindComponent<OpenModel>(), Throws.Nothing);
                 Assert.That(() => this.messageBus.SendMessage(new DomainChangedEvent(null, null)), Throws.Nothing);
             });

--- a/COMET.Web.Common/COMET.Web.Common.csproj
+++ b/COMET.Web.Common/COMET.Web.Common.csproj
@@ -50,7 +50,7 @@
     <PackageReference Include="Microsoft.AspNetCore.WebUtilities" Version="10.0.7" />
     <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="10.0.7" />
     <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="10.0.7" />
-    <PackageReference Include="ReactiveUI" Version="23.1.8" />
+    <PackageReference Include="ReactiveUI" Version="23.2.28" />
     <PackageReference Include="System.Drawing.Common" Version="10.0.7" />
     <PackageReference Include="System.Linq.Dynamic.Core" Version="1.7.2" />
     <PackageReference Include="System.Text.RegularExpressions" Version="4.3.1" />

--- a/COMETwebapp/COMETwebapp.csproj
+++ b/COMETwebapp/COMETwebapp.csproj
@@ -34,7 +34,7 @@
         <PackageReference Include="System.Drawing.Primitives" Version="4.3.0" />
         <PackageReference Include="Z.Blazor.Diagrams" Version="3.0.4.1" />
 
-        <PackageReference Include="ReactiveUI.Blazor" Version="23.1.8" />
+        <PackageReference Include="ReactiveUI.Blazor" Version="23.2.28" />
         <PackageReference Include="ReactiveUI.SourceGenerators" Version="2.6.1">
             <PrivateAssets>all</PrivateAssets>
             <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>

--- a/COMETwebapp/Dockerfile
+++ b/COMETwebapp/Dockerfile
@@ -5,10 +5,6 @@ FROM mcr.microsoft.com/dotnet/sdk:10.0 AS build
 ARG DEVEXPRESS_NUGET_KEY
 ARG PACKAGE_TOKEN
 
-# Workaround for NU3012: the ReactiveUI/Splat NuGet signing certificate has been revoked.
-# Disable enforced package signature verification until the packages are updated. See issue for the long-term fix.
-ENV DOTNET_NUGET_SIGNATURE_VERIFICATION=false
-
 COPY COMET.Web.Common COMET.Web.Common
 COPY COMETwebapp COMETwebapp
 

